core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -638,6 +652,43 @@ static Parameter ofRequestObject(Class<?> type,
             this.name = name;
             this.defaultValue = defaultValue;
             this.requestConverterFunction = requestConverterFunction;
+
+            if (type.isEnum()) {
+                final Set<? extends Enum<?>> enumInstances = EnumSet.allOf(type.asSubclass(Enum.class));
+                Map<String, ? extends Enum<?>> upperCaseEnumMap = enumInstances.stream().collect(
+                        toImmutableMap(e -> Ascii.toUpperCase(e.name()), Function.identity(), (e1, e2) -> e1));
+                if (enumInstances.size() != upperCaseEnumMap.size()) {
+                    upperCaseEnum = null;
```

Comment:
- How about building a lower-case map given most HTTP API uses lower-case values?
- Instead of setting the map to null, how about building a map with case-sensitive enum values, so that `getEnumElement()` becomes simpler?

Reply:
Checked & Updated Thanks!!